### PR TITLE
Feature/refactor

### PR DESCRIPTION
**Pull request by ladydascalie**
These are just all the changes that we'd made but had not merge requested until then.
For posterity they should be included, (and because they break `brew install` if we don't)

### CODE REVIEW

File: domain/domain.go [hunk 1]
```diff
@@ -16,3 +18,11 @@ func HttpStatusFmt(code int) string {
 	}
 	return fmt.Sprintf("%d (%s)", code, txt)
 }
+
+// Equals performs a deep equals against two objects and
+// fails if they're not equal.
+func Equals(tb testing.TB, exp interface{}, got interface{}) {
+	if !reflect.DeepEqual(exp, got) {
+		tb.Fatalf("\texp: %#v\n\tgot: %#v", exp, got)
```
Comment (codingconcepts) [inline domain/domain.go:26]: Perhaps use the format we're using elsewhere that shows the value and the type?

File: domain/getter_header.go [hunk 2]
```diff
@@ -1,19 +1,19 @@
...
 
 // HeaderGetter extracts information from response headers.
 type HeaderGetter struct{}
 
 // Get extracts a value out of request headers.
-func (e *HeaderGetter) Get(c format.GetterConfig, header http.Header) (value string, err error) {
+func (e *HeaderGetter) Get(path string, header http.Header) (value string, err error) {
 	for k, v := range header {
-		if c.Path == k && len(v) > 0 {
+		if strings.ToLower(path) == strings.ToLower(k) && len(v) > 0 {
```
Comment (codingconcepts) [inline domain/getter_header.go:16]: You can use `strings.EqualFold` here instead.

File: domain/litmus_test.go [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+	var tf TestFile
+	if err := toml.Unmarshal(b, &tf); err != nil {
+		t.Fatal(err)
+	}
+
+	spew.Dump(tf)
+}
+
+//func TestEnvFormat(t *testing.T) {
```
Comment (codingconcepts) [inline domain/litmus_test.go:24]: I remember us commenting this while we were breaking stuff but it'll need to be uncommented (obvs).

File: examples/1_type_test.toml [hunk 4]
```diff
@@ -1,61 +1,64 @@
...
-[[litmus.test.getters]]
-type="body"
-path="gzipped"
-exp="true"
-[[litmus.test.getters]]
-type="head"
-path="Content-Type"
-exp="application/json"
-
+#[litmus]
```
Comment (codingconcepts) [inline examples/1_type_test.toml:1]: Same as commenting code above.

File: domain/domain.go (hunk 1 above)
Comment (ladydascalie) [reply domain/domain.go:26]: done

File: domain/getter_header.go (hunk 2 above)
Comment (ladydascalie) [reply domain/getter_header.go:16]: good shout, done

File: examples/1_type_test.toml (hunk 4 above)
Comment (ladydascalie) [reply examples/1_type_test.toml:1]: this whole file needs a complete re-write

File: domain/litmus_test.go (hunk 3 above)
Comment (ladydascalie) [reply domain/litmus_test.go:24]: this whole file needs a complete re-write

**Pull request merged**